remapping_commands.sh

```diff
@@ -0,0 +1,189 @@
+rm temp.txt
+
+
+
+
+
+
+#2) Mapping with bowtie2
+echo "---------------------------2) Mapping with bowtie2---------------------------"
+# index the target assembly: longest step
+#"index" is the name of the output files, not a directory
+echo "#### Building index: may take a while ####"
+bowtie2-build $newgenome index
+echo "#### Mapping results: ####"
+bowtie2 -f -x index variant_reads.out.fa | samtools view -bS - > reads_aligned.bam
+#example output:
+#>1094 reads; of these:
+#>  1094 (100.00%) were unpaired; of these:
+#>   7 (0.64%) aligned 0 times
+#>    48 (4.39%) aligned exactly 1 time
+#>    1039 (94.97%) aligned >1 times
+#>99.36% overall alignment rate
+
+
+
+
+
+#3) Data extraction
+echo '------------------------------3) Data extraction-----------------------------'
+#extract chromosome, rsID, position on new genome, original variant allele, qual, filter and info fields
+#samtools view "$species"_reads_aligned.bam | awk '($1 ~ "|") {split($1,info,"|"); print $3"\t"$4+50"\t"info[4]"\t"info[3]"\t"info[5]"\t"info[6]"\t"info[7]}' > variants_remapped.vcf
+samtools sort reads_aligned.bam -o reads_aligned.sorted.bam
+samtools index reads_aligned.sorted.bam
+./reverse_strand.py -i reads_aligned.sorted.bam -o variants_remapped.vcf
+rm reads_aligned.bam
+
+#delete missing chromosomes (*)
+#awk '($1 != "*") {print $0}' variants_remapped.vcf > variants_remapped.fixed.vcf
+
+
+#add interval for variant position in bed format (required by getfasta)
+awk '{print $1"\t"$2-1"\t"$2"\t"$3"\t"$4"\t"$5"\t"$6"\t"$7}' variants_remapped.vcf > variants_remapped.bed
+
+#getfasta to get the genome alleles and then extract the genome alleles without the positions
+bedtools getfasta -tab -fi $newgenome -bed variants_remapped.bed | awk '{print $2}'> genome_alleles.txt
+rm variants_remapped.bed
+#make all alleles caps (sometimes there are lower-case bases)
+tr a-z A-Z < genome_alleles.txt > genome_alleles.fixed.txt
+
+#paste all the information into the correct columns
+awk '{print $1"\t"$2"\t"$3}' variants_remapped.vcf > temp_var1.txt
+awk '{print $4"\t"$5"\t"$6"\t"$7}' variants_remapped.vcf > temp_var2.txt
+paste temp_var1.txt genome_alleles.fixed.txt temp_var2.txt > var_pre_final.vcf
+rm temp_var1.txt temp_var2.txt genome_alleles.txt variants_remapped.vcf
+
+#header:
+#create list of contigs/chromosomes to be added to the header
+awk '{print $1}' var_pre_final.vcf | sort | uniq > contig_names.txt
+while read CHR; do echo "##contig=<ID=$CHR>"; done < contig_names.txt > contigs.txt
+rm contig_names.txt
+
+#add the reference assembly
+echo "##reference="$newgenomeaccession >> contigs.txt
+
+#copy everything that isn't ##contig or ##reference from the old header to a temp
+awk '($1 !~ /^##contig/ && $1 !~ /^##reference/) {print $0}' vcf_header.txt > temp_header.txt
+rm vcf_header.txt
+
+#find the end of the INFO tags and print the contigs (using custom python script)
+chmod a+x write_header.py
+./write_header.py -i temp_header.txt -c contigs.txt -o final_header.txt
```

Comment:
If `vcf_header.txt` was created using only lines starting with `##`, then this Python script might not even be necessary because getting the final header would be just a matter of concatenating a bunch of other files.

Reply:
See [other comment.](https://github.com/EBIvariation/variant-remapping/pull/1/files#r308158748)